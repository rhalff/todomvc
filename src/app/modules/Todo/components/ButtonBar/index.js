// @flow
import React, { PureComponent } from 'react'

import FilterButton from '../FilterButton/index'
import ListItem from '../ListItem/index'
import styles from './scss/Buttonbar.scss'

import type { Buttons } from '../../types/buttons'

type Props = {
  /**
   * List of buttons
   *
   *
   */
  buttons: Buttons,
  filter: string,
  onClick: Function
}

class ButtonBar extends PureComponent<Props> {
  render() {
    const { buttons, filter, onClick } = this.props

    return (
      <ul className={styles.buttonBar}>
        {buttons.map(button => (
          <ListItem onClick={onClick} key={button.name} name={button.name}>
            <FilterButton {...button} active={button.name === filter} />
          </ListItem>
        ))}
      </ul>
    )
  }
}

export default ButtonBar
