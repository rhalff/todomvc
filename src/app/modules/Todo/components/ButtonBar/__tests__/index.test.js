// @flow
import React from 'react'

import renderer from 'react-test-renderer'

import ButtonBar from '../index'

const buttons = [
  { name: 'ALL', active: true },
  { name: 'ACTIVE' },
  { name: 'COMPLETED' }
]

test('Can show Buttonbar', () => {
  const component = renderer.create(<ButtonBar buttons={buttons} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
