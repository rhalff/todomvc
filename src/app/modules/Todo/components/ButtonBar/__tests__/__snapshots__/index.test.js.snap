// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show Buttonbar 1`] = `
<ul
  className="buttonBar"
>
  <li
    className="listItem"
    onClick={[Function]}
  >
    <a
      className="filterButton"
      href="/#ALL"
      name="ALL"
    >
      ALL
    </a>
  </li>
  <li
    className="listItem"
    onClick={[Function]}
  >
    <a
      className="filterButton"
      href="/#ACTIVE"
      name="ACTIVE"
    >
      ACTIVE
    </a>
  </li>
  <li
    className="listItem"
    onClick={[Function]}
  >
    <a
      className="filterButton"
      href="/#COMPLETED"
      name="COMPLETED"
    >
      COMPLETED
    </a>
  </li>
</ul>
`;
